### PR TITLE
Avoid passing enkf_main_type to enkf_main_inflate()

### PR DESCRIPTION
**Pull request by BjarneHerland**
Replace enkf_main_type* with ensemble_config_type* and ensamble_size which contains sufficient info for the code-path starting at enkf_main_inflate()

### CODE REVIEW

File: libres/lib/enkf/enkf_main.cpp [hunk 1]
```diff
@@ -341,12 +340,12 @@ static void enkf_main_node_std(const vector_type *ensemble,
...
-void enkf_main_inflate_node(enkf_main_type *enkf_main, enkf_fs_type *src_fs,
-                            enkf_fs_type *target_fs, int report_step,
+void enkf_main_inflate_node(const ensemble_config_type *ens_config,
+                            enkf_fs_type *src_fs, enkf_fs_type *target_fs,
+                            const int ens_size, int report_step,
                             const char *key, const enkf_node_type *min_std) {
-    int ens_size = enkf_main_get_ensemble_size(enkf_main);
     vector_type *ensemble = enkf_main_alloc_node_ensemble(
-        enkf_main, src_fs, key, report_step); // Was ANALYZED
+        ens_config, ens_size, src_fs, key, report_step); // Was ANALYZED
```
Comment (dafeda) [inline libres/lib/enkf/enkf_main.cpp:348]: Think we can get rid of the commend // Was ANALYZED

**Comment by dafeda** [comment]
There's a comment above `enkf_main_inflate` that says:

/*
    Denne burde istedet loope gjennom noklene fra use_count
    direkte.
*/

Does this make sense to you?

Not directly related to this PR, but what is `inflation`?

File: libres/lib/enkf/enkf_main.cpp (hunk 1 above)
Comment (BjarneHerland) [reply libres/lib/enkf/enkf_main.cpp:348]: Yeah - will do

**Comment by BjarneHerland** [comment]
> There's a comment above `enkf_main_inflate` that says:
> 
> /* Denne burde istedet loope gjennom noklene fra use_count direkte. */
> 
> Does this make sense to you?

Current code loops over all possible keys, and for each key checks if `use_count` contains it before doing its _thang_. The more elegant (and efficient) implementation is to just iterate over existing keys in `use_count`, avoiding a number of iterations plus a lot of lookups.

**Review by dafeda — APPROVED** [review]
Thanks for the explanation, this looks good.

**Comment by BjarneHerland** [comment]
test this please

**Pull request merged**